test/grid/structured_grid.hpp

```diff
@@ -106,16 +111,18 @@ class StructuredGrid {
             const auto incremented = [] (auto pos, int dir) { pos[dir]++; return pos; };
             std::vector point_ids{cell_pos};
             point_ids.push_back(incremented(cell_pos, 0));
-            point_ids.push_back(incremented(incremented(cell_pos, 0), 1));
-            point_ids.push_back(incremented(cell_pos, 1));
+            if constexpr (dim == 2) {
+                point_ids.push_back(incremented(incremented(cell_pos, 0), 1));
+                point_ids.push_back(incremented(cell_pos, 1));
+            }
             if constexpr (dim == 3) {
                 point_ids.push_back(incremented(cell_pos, 2));
                 point_ids.push_back(incremented(incremented(cell_pos, 0), 2));
                 point_ids.push_back(incremented(incremented(incremented(cell_pos, 0), 1), 2));
                 point_ids.push_back(incremented(incremented(cell_pos, 1), 2));
             }
 
-            _cell_corner_indices[cell.id].resize(dim == 2 ? 4 : 8);
+            _cell_corner_indices[cell.id].resize(std::pow(2, dim));
```

Comment:
```suggestion
            _cell_corner_indices[cell.id].resize(1 << dim);
```

since `pow` operated on floating point numbers.

Reply:
true, I kind of assumed there would be an overload for integral types... Interesting that the pipeline did not complain even with pedantic settings...